source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs

```diff
@@ -61,6 +61,8 @@ public void ModifyFile(string filePath, IVariables variables)
                 {
                     var selectedNodes = navigator.XPath2SelectNodes(xPathExpression, nsManager);
                     var variableValue = variables.Get(variable.Key);
+                    if (variableValue == null)
+                        continue;
```

Comment:
How do we want to handle empty strings here?

Reply:
good question. I've revisited and put it back to assigning the nulls, the Xml objects all seems to deal with that correctly and then we don't risk accidentally changing behavior